src/parsers.ts

```diff
@@ -8,7 +8,8 @@ export type DocumentUri = string;
 export type Segment = string | number;
 export type JsonPath = Segment[];
 
-export interface IParserResult<T = any, A extends object = object, L = unknown> extends IParserASTResult<T, A, L> {
+export interface IParserResult<T = any, A extends object = object, L = unknown, M = unknown>
+  extends IParserASTResult<T, A, L, M> {
```

Comment:
TALM — maybe describing these parameters would be helpful? I would never be able to get it.

Reply:
<img width="135" alt="image" src="https://user-images.githubusercontent.com/1416224/62208714-f007fe00-b397-11e9-9a62-5ade302aca92.png">

Reply:
Types
AST
Linemap
Metadata